sql/table.go

```diff
@@ -273,12 +272,20 @@ func encodeSecondaryIndexes(tableID structured.ID, indexes []structured.IndexDes
 	var secondaryIndexEntries []indexEntry
 	for _, secondaryIndex := range indexes {
 		secondaryIndexKeyPrefix := encodeIndexKeyPrefix(tableID, secondaryIndex.ID)
-		secondaryIndexKey, err := encodeIndexKey(secondaryIndex, colMap, values, secondaryIndexKeyPrefix)
+		secondaryIndexKey, err := encodeIndexKey(secondaryIndex.ColumnIDs, colMap, values, secondaryIndexKeyPrefix)
 		if err != nil {
 			return nil, err
 		}
 
 		if secondaryIndex.Unique {
+			// Append the primaryIndexKeySuffix if any column is NULL.
+			for _, id := range secondaryIndex.ColumnIDs {
+				if _, ok := colMap[id]; !ok {
```

Comment:
Is that the right check? I think that works if the column is `NULL` because it wasn't part of the columns being inserted into. But what if an explicit NULL is being inserted. I think it would be somewhat safer if `encodeIndexKey` returned a bool indicating whether any of the values was `NULL`.